src/common/util.c

```diff
@@ -229,3 +230,30 @@ char *oscap_expand_ipv6(const char *input)
 
 	return ret;
 }
+
+char *oscap_path_join(const char *path1, const char *path2)
+{
+	if (path1 == NULL) {
+		return oscap_strdup(path2);
+	}
+	if (path2 == NULL) {
+		return oscap_strdup(path1);
+	}
+	size_t path1_len = strlen(path1);
+	size_t path2_len = strlen(path2);
+	size_t path2_shift = 0;
+	while (path1_len >= 1 && path1[path1_len - 1] == PATH_SEPARATOR) {
+		path1_len--;
+	}
+	while (path2_shift < path2_len && path2[path2_shift] == PATH_SEPARATOR) {
+		path2_shift++;
+	}
+	path2_len -= path2_shift;
+	const size_t joined_path_len = path1_len + 1 + path2_len;
```

Comment:
this is a nitpick but for readability I would except the +1 to be at the end because it is for the \0 terminator which is at the end.